Automattic/custom-fonts
# Fix linting issues

This PR addresses some long-overdue maintenance on this project, which goes well beyond the original linting issues that came up when I deployed v1.1.2 to WordPress.com in D85434-code.

The PR includes the following changes:
 * Address some linting issues in the testing code
 * Bump the `node` version in `.nvmrc` to `16.13.x`
 * Bump a number of npm dependencies using `npm audit fix` and `npm audit fix --force` - we had a _lot_ of vulnerabilities in our dependencies, including a few critical ones
 * Allow PHPUnit 7 or 8, so we can support PHP 8; this required adding some type declarations to the test code
 * Replace uses of the deprecated `WP_Mock::wpFunction()` with `WP_Mock::userFunction()`
 * Update to the most recent version of `uglify-js` to get `npm run dist` working
 * Bump the version to 2.0.0 to reflect the big dependency jumps
 * Update the README to capture the errors from `fvd` and note that a GitHub release is needed to make the package available via Composer

### Testing instructions

* Checkout this branch locally
* Assuming you have `nvm` installed, run `nvm use` from the base directory
* Run `npm install`
* Verify that everything installs cleanly (with the exception of `fvd`)
* Run `npm run test`
* Verify that the tests complete successfully
* Run `npm run build`
* Verify that the build completes successfully
* Run `npm run watch`
* Verify that the watcher is working and reacts to changes in any `src/js/*.js` files
* Run `npm run dist`
* Verify that the build completes successfully and updates the `js/jetpack*.js` files
* Upload the updated JS and PHP files to an Atomic or Simple site
* Verify that you can still interact with the _Fonts_ section of your Customizer
* Verify that you can still save alternative fonts for your site -- ensure that all the customizer changes are saved
* Verify that the selected fonts appear in the front-end of your site
